public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -151,54 +153,56 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
     // replace templated variables
     allQueries = this.templateSrv.replace(allQueries, scopedVars);
 
-    return this._seriesQuery(allQueries, options).then((data: any): any => {
-      if (!data || !data.results) {
-        return [];
-      }
-
-      const seriesList = [];
-      for (i = 0; i < data.results.length; i++) {
-        const result = data.results[i];
-        if (!result || !result.series) {
-          continue;
+    return this._seriesQuery(allQueries, options)
+      .toPromise()
+      .then((data: any): any => {
+        if (!data || !data.results) {
+          return [];
```

Comment:
I think classicQuery should return an Observable instead and then we could remove __from__ creator here https://github.com/grafana/grafana/blob/316a7997a53882fe6acb37f4cffa0757bb674012/public/app/plugins/datasource/influxdb/datasource.ts#L80

and then we don't need to do `toPromise` here. WDTY?